doc/nrf/ug_bootloader.rst

```diff
@@ -92,7 +92,96 @@ These signed variants can be used to perform an upgrade procedure through the :r
 Adding a bootloader chain to your application
 *********************************************
 
-See :ref:`bootloader_build_and_run` in the documentation of the :ref:`bootloader` sample for instructions on how to add an immutable bootloader as first stage of the secure bootloader chain.
+The |NCS| includes a sample implementation of an immutable bootloader.
+As upgradable bootloader, it provides and extends :doc:`mcuboot:index`.
 
-See :doc:`mcuboot:index` for information on how to implement an upgradable bootlader.
+Both bootloaders can easily be included in your application using :ref:`ug_multi_image`.
+
+Adding an immutable bootloader
+==============================
+
+To add an immutable bootloader to your application, set :option:`CONFIG_SECURE_BOOT` and add your private key file under :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`.
+|how_to_configure|
+
+See the documentation of the :ref:`bootloader` sample for more information.
+The :ref:`bootloader_build_and_run` section has detailed instructions for adding the immutable bootloader as first stage of the secure bootloader chain.
+
+Adding an upgradable bootloader
+===============================
+
+To add an upgradable bootloader to your applications, set :option:`CONFIG_BOOTLOADER_MCUBOOT`.
+|how_to_configure|
+
+The upgradable bootloader should always be used in combination with the immutable bootloader.
```

Comment:
This statement is maybe too forceful. To me this indicates that you'll always need both, when in reality it really depends on your use case. You may maybe just want the immutable bootloader or as stated bellow just MCUBoot without having it upgradable.

Reply:
So what are the use cases? Using only the immutable bootloader is straight-forward (and I didn't mean that here), but when and why would you want to use an ugradable bootloader that cannot be upgraded?

Reply:
To upgrade your application if you don't care about trust or security. But if we take the abstract view that this is any "upgradable bootloader" I think your point is valid.